src/promdens/promdens.py

```diff
@@ -172,34 +171,50 @@ def wigner_transform(self, tprime, de):
         :param de: excitation energy (a.u.)
         :return: Wigner pulse transform
         """
-        # setting an adaptive integration step according to the frequency of the integrand oscillations (de - omega)
+
         loc_omega = self.omega + 2*self.lchirp*tprime
-        # NOTE: Assuming de is in atomic units where hbar=1
-        # If de == loc_omega, we're in resonance and there are no oscillations,
-        # integrate only the envelope intensity so we can set a larger integration step
-        if de == loc_omega:
-            T = 2*np.pi/loc_omega
-        else:
-            T = 2*np.pi/(np.min([np.abs(de - loc_omega), loc_omega]))
-        ds = np.min([T/50, self.fwhm/500])  # time step for integration
-
-        # integration ranges for different pulse envelopes
-        # ideally, we would integrate from -infinity to infinity, yet this is not very computationally efficient
-        # empirically, it was found out that efficient integration varies for different pulses
-        # analytic formulas should be implemented in the future to avoid that
-        factor = {
-            'gauss': 7.5,
-            'lorentz': 50,
-            'sech': 20,
-            'sin': 3,
-            'sin2': 4,
-        }
-
-        # instead of calculating the complex integral int_{-inf}^{inf}[E(t+s/2)E(t-s/2)exp(i(w-de)s)]ds we use the
-        # properties of even and odd fucntions and calculate 2*int_{0}^{inf}[E(t+s/2)E(t-s/2)cos((w-de)s)]ds
-        s = np.arange(0, factor[self.envelope_type]*self.fwhm, step=ds)
-        cos = np.cos((de - loc_omega)*s)
-        integral = 2*np.trapz(x=s, y=cos*self.calc_field_envelope(tprime + s/2)*self.calc_field_envelope(tprime - s/2))
+        # analytic integrals if available
+        if self.envelope_type == 'gauss':
+            integral = 16**(-(tprime - self.t0)**2/self.fwhm**2)*np.exp(-self.fwhm**2*(de - loc_omega)**2/np.log(16))*self.fwhm*np.sqrt(
```

Comment:
To make this more readable, perhaps you could extract the prefactor to a separate variable

```suggestion
            prefactor = 16**(-(tprime - self.t0)**2/self.fwhm**2)*self.fwhm*np.pi/np.log(2)
            integral = prefactor * np.exp(-self.fwhm**2*(de - loc_omega)**2/np.log(16))
```

Reply:
Yes, that would make sense.